generate-test-vector.py

```diff
@@ -57,6 +57,16 @@ def get_p2tr_witness(priv_key):
     sig = priv_key.sign_schnorr(msg).hex()
     return serialize_witness_stack([sig])
 
+def get_leafp2tr_witness(priv_key, internal_key, add_annex=True):
+    msg = hashlib.sha256(b'message').digest()
+    sig = priv_key.sign_schnorr(msg).hex()
+    leaf_script = (0x20 << 264) | (int.from_bytes(priv_key.get_pubkey().get_bytes(True), 'big') << 8) | 0xac
+    control = (0xc1 << 512) | (internal_key << 256) | int.from_bytes(hashlib.sha256(b'leafhash').digest(), 'big')
```

Comment:
I don't think control block should contain the hash.
IIUC the leaf here is the root of taptree, so it should be the 33 bytes.

Reply:
You can use `tap` command from btcdeb to verify your calculations

Reply:
I extracted the witness stack from existing taproot transactions I had. I'll update this and tweak the scriptPubKey with the script tree merkle root.